### PR TITLE
Fix underline behavior in Link component

### PR DESCRIPTION
**Pull request by camertron**
### What are you trying to accomplish?

The `Primer::Beta::Link` component does not render an underlined link when the `underline: true` argument is passed. I'm not sure why, since links are underlined by default. This PR makes the underlining explicit and stops using the primer/css `.no-underline` utility class. It also sets `underline: false` as the default, since that's the current behavior anyway, and will likely result in sweeping (and potentially unwanted) changes to dotcom.

### Screenshots

|Before|After|
|-|-|
|A screenshot of the Link component rendered in a Lookbook preview. The link is not underlined. The params panel is open at the bottom showing the 'Underline' parameter is flipped on.|A screenshot of the Link component rendered in a Lookbook preview. The link is underlined. The params panel is open at the bottom showing the 'Underline' parameter is flipped on.|

### Integration

No changes necessary in production.

#### Risk Assessment
  

- [x] **Low risk** the change is small, highly observable, and easily rolled back.
- [ ] **Medium risk** changes that are isolated, reduced in scope or could impact few users. The change will not impact library availability.
- [ ] **High risk** changes are those that could impact customers and SLOs, low or no test coverage, low observability, or slow to rollback.

### Accessibility

- **No new axe scan violation** - This change does not introduce any new [axe scan](https://thehub.github.com/epd/engineering/dev-practicals/frontend/accessibility/readiness-routine/development/#axe-scans) violations.

Underlining links when they are surrounded by other text is considered an accessibility best-practice. So, while this change does not fix any axe violations, it is more accessible and may impact teams' scorecards positively.

### Merge checklist

- [x] Added/updated tests
~- [ ] Added/updated documentation~
~- [ ] Added/updated previews (Lookbook)~
- [x] Tested in Chrome
- [ ] Tested in Firefox
- [ ] Tested in Safari
- [ ] Tested in Edge

Take a look at the [What we look for in reviews](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#what-we-look-for-in-reviews) section of the contributing guidelines for more information on how we review PRs.

### CODE REVIEW

File: app/components/primer/beta/link.rb [hunk 1]
```diff
@@ -47,7 +47,7 @@ def initialize(href: nil, scheme: DEFAULT_SCHEME, muted: false, underline: true,
           SCHEME_MAPPINGS[fetch_or_fallback(SCHEME_MAPPINGS.keys, scheme, DEFAULT_SCHEME)],
           "Link",
           "Link--muted" => muted,
-          "no-underline" => !underline
+          "Link--underline" => underline
```
Comment (jonrohan) [inline app/components/primer/beta/link.rb:50]: This seems like all links will always be underlined? Should we default `underline` to false then?

Comment (camertron) [reply app/components/primer/beta/link.rb:50]: I think underlining is the HTML default, which means we should default to `underline: true` also.

**Review by jonrohan — APPROVED** [review]
🤘🏻 Lovely fix

**Comment by lesliecdubs** [comment]
Thanks for the report @kintner! I've filed a follow-up issue to track resolution: https://github.com/primer/view_components/issues/2277

**Pull request merged**